text/0011-add-new-path-to-account-creation-on-asset-hubs.md

```diff
@@ -0,0 +1,167 @@
+    } else {
+        // The destination does not exist and the user has opted in to create it via a swap.
+        //
+        // We will try to swap the asset provided for the existential deposit, depositing the ED in
+        // the destination account. If the asset does not have an Asset Conversion pair with DOT or
+        // the asset amount isn't enough to acquire the existential deposit, this will fail. But we
+        // generally think (a) pairs will exist, and (b) the ED is small and UIs can easily verify
+        // that this should succeed, so failures should be rare.
+        //
+        // The swap returns the amount of the asset consumed to acquire the ED.
+        let consumed = Swap::swap_tokens_for_exact_tokens(
+            from,                // sender
+            vec![asset, dot],    // path, where `dot` is Multilocation {parents: 1, interior: Here}
+            existential_deposit, // amount_out, we need the ED for the account
+            destination,         // send_to
+            ..
+        )?;
+        // We used some asset for the swap, so we have to subtract that from the amount.
+        let remaining_asset_amount = amount.saturating_sub(consumed);
+        // Now we transfer whatever amount is left, knowing that the destination account exists.
+        // This could still fail if the remaining amount is less than the minimum balance required
+        // by the asset class.
+        Self::do_transfer(asset, from, destination, remaining_asset_amount, ..)?;
+    }
+}
+```
+
+## Drawbacks
+
+This solution would automatically convert some amount of another asset to DOT when acquiring DOT
+was perhaps not the recipient's intent. However, this is opt-in.
+
+## Testing, Security, and Privacy
+
+An attacker that wanted to bloat state by sending worthless assets to many new accounts would need
+to put the DOT into an Asset Conversion pool with the asset (thereby making the asset _not_
+worthless with respect to DOT). This would provide the same cost and economic security as just
+sending the existential deposit of DOT to all the new accounts. This approach is no less secure
+than the DOT-only existential deposit system.
+
+This proposal introduces no privacy enhancements or reductions.
+
+## Performance, Ergonomics, and Compatibility
+
+### Performance
+
+The function to transfer assets will need to charge a larger weight at dispatch to account for the
+possibility of needing to perform a swap for DOT. It could return any unused weight.
+
+The implementation could also include witness data as to the destination account's existence so
+that the block builder can appropriately budget for the weight.
+
+### Ergonomics
+
+This proposal would benefit the ergonomics of the system for end users by allowing all assets to
+create destination accounts when needed.
+
+### Compatibility
+
+This change would require changes to the Assets pallet to add the new account creation path.
+
+## Prior Art and References
+
+Discussions with:
```

Comment:
Acala/Karura have the feature to allow assets to pay for tx fee with help of swap and ensure there is ED amount of the native token for sufficiency
https://github.com/AcalaNetwork/Acala/blob/master/modules/transaction-payment/src/lib.rs